src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -44,7 +44,19 @@ public AuthenticationHandler(bool preAuthenticate, ICredentials credentials, Htt
 
             HttpResponseMessage response = await _innerHandler.SendAsync(request, cancellationToken).ConfigureAwait(false);
 
-            if (!_preAuthenticate && response.StatusCode == HttpStatusCode.Unauthorized)
+            // In case of redirection, ensure _credentials as CredentialCache
+            if (response.StatusCode == HttpStatusCode.MultipleChoices ||
+                response.StatusCode == HttpStatusCode.Moved ||
+                response.StatusCode == HttpStatusCode.Found ||
+                response.StatusCode == HttpStatusCode.SeeOther ||
+                response.StatusCode == HttpStatusCode.TemporaryRedirect)
```

Comment:
Yeah, I took these from the AutoRedirectHandler. There are in a switch statement on the autoredirect handler, the helper pattern could be something like having these status codes as HttpRedirectionStatusCode.{variousoptions}, but I don't see value in it, adds more confusion. Do you have ideas?

Reply:
> Do you have ideas?

Hmm.  I guess it's fine for now.  Consider just adding a comment in both places that if one changes the other should change as well.  We should also look at consolidating these handlers, if for no other reason than to avoid the extra level of async methods / tasks, but it'd potentially also help bring this shared logic closer together.